sig-architecture/charter.md

```diff
@@ -1,40 +1,175 @@
+    <td>API conventions, API linter</td>
+  </tr>
+  <tr>
+    <td>Deprecation policy</td>
+    <td>Maintaining, evolving, and enforcing the deprecation policy</td>
+    <td>Deprecation policy</td>
+  </tr>
+  <tr>
+    <td>Code organization</td>
+    <td>Overall code organization, including github repositories and branching methodology, top-level and pkg OWNERS of kubernetes/kubernetes, kubernetes-template-project</td>
+    <td>kubernetes-template-project</td>
+  </tr>
+  <tr>
+    <td>KEP</td>
+    <td>Develop and drive technical enhancement review process</td>
+    <td>Process, Template</td>
+  </tr>
+</table>
+
+
+## Roles
+
+The following roles are required for the SIG to properly function. In the event that any role is unfilled, the SIG will make a best effort to fill it, and any decisions reliant on a missing role will be postponed until the role is filled.
+
+* **Chair**
+
+    * Number: 2-3
+
+    * Run operations and processes governing the SIG
+
+    * An initial set of chairs was established at the time the SIG was founded.
+
+    * Chairs *MAY* decide to step down at anytime and propose a replacement, who must be approved by all of the other chairs. This *SHOULD* be supported by a majority of SIG Members.
+
+    * Chairs *MAY* select additional chairs by consensus. This *SHOULD* be supported by a majority of SIG Members.
+
+    * Chairs *MUST* remain active in the role and are automatically removed from the position if they are unresponsive for > 3 months and *MAY* be removed by consensus of the other Chairs and all of the Technical leads if not proactively working with other Chairs to fulfill responsibilities.
+
+    * A majority of chairs cannot be from a single company. If a chair’s change of company affiliation causes this threshold to be exceeded, one of the chairs must step down and propose a replacement, as per the procedure above.
+
+    * Defined in[ sigs.yaml](https://github.com/kubernetes/community/blob/master/sigs.yaml#L1454)
+
+* SIG Technical Leads
+
+    * Number: 3
+
+    * Establish new subprojects, and retire existing subprojects
+
+    * Resolve cross-subproject technical issues and decisions, and escalations from subprojects
+
+    * Decision-making MUST be by consensus. It’s particularly important for the technical leads to provide cohesive technical guidance for the project as a whole.
+
+    * The initial set of technical leads is a subset of the long-standing group of API approvers: Clayton Coleman, Tim Hockin, and Brian Grant
+
+    * Technical leads MUST have a demonstrated mastery, proficiency, and review history in the project sufficient to assess their ability to arbitrate highly technical and project-wide decisions
+
+    * Technical leads *MUST* remain active in the role and are automatically removed from the position if they are unresponsive for > 3 months
```

Comment:
More clarity on what "unresponsive" means. As written, simply replying to an email might count as "responsive" but I don't think that's what we want/mean.